### PR TITLE
[V0.96.8] carry ressource crates

### PR DESCRIPTION
**Pull request by Dubjunk**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes  |
| Needs wipe? | no  |

### Description:

This PR adds the ability to carry ressource crates without using ACE.

### Content:
- [x] Ressource crate carrying

### Successfully tested on:
- [x] Local MP
- [ ] Dedicated MP

### CODE REVIEW

File: Missionframework/functions/fn_addActionsPlayer.sqf [hunk 1]
```diff
@@ -370,4 +370,20 @@ _player addAction [
     "
 ];
 
+// Drop crate
+_player addAction [
+    "<t color='#FFFF00'>", localize "STR_ACTION_CRATE_DROP", "</t>"] joinString "",
```
Comment (Wyqer) [inline Missionframework/functions/fn_addActionsPlayer.sqf:375]:
```suggestion
    ["<t color='#FFFF00'>", localize "STR_ACTION_CRATE_DROP", "</t>"] joinString "",
```

File: Missionframework/scripts/client/ammoboxes/ammobox_action_manager.sqf [hunk 2]
```diff
@@ -57,10 +57,12 @@ while {true} do {
...
                 _b_action_id3 = _next_box addAction ["<t color='#FFFF00'>" + localize "STR_ACTION_CRATE_VALUE" + "</t>",{[_this select 0] call KPLIB_fnc_checkCrateValue;uiSleep 3; hint "";},"",-503,true,true,"","build_confirmed == 0 && (_this distance _target < 5) && (vehicle player == player)"];
                 _b_action_id4 = _next_box addAction ["<t color='#FFFF00'>" + localize "STR_ACTION_CRATE_PUSH" + "</t>",{(_this select 0) setPos ((_this select 0) getPos [1, (player getDir (_this select 0))]);},"",-504,true,false,"","build_confirmed == 0 && (_this distance _target < 5) && (vehicle player == player)"];
+                _b_action_id5 = _next_box addAction ["<t color='#FFFF00'>" + localize "STR_ACTION_CRATE_CARRY" + "</t>",{(_this select 0) attachTo [(_this select 1), [0, 2, 1]];},"",-504,true,false,"","build_confirmed == 0 && (_this distance _target < 5) && (vehicle player == player) && (isNull attachedTo _target)"];
```
Comment (Wyqer) [inline Missionframework/scripts/client/ammoboxes/ammobox_action_manager.sqf:60]: Having this added here is "fine enough" atm. Basically the whole ammobox_action_manager should be removed in the upcoming changes.
Until then it's fine. Wouldn't make sense to refactor the ammobox action manager in the scope of the carry action.

File: Missionframework/functions/fn_addActionsPlayer.sqf [hunk 3]
```diff
@@ -370,4 +370,20 @@ _player addAction [
...
+    {detach ((attachedObjects (_this select 1)) select 0)},
```
Comment (Wyqer) [inline Missionframework/functions/fn_addActionsPlayer.sqf:376]:
```suggestion
    {detach ((attachedObjects player) select 0)},
```

File: Missionframework/functions/fn_addActionsPlayer.sqf [hunk 4]
```diff
@@ -370,4 +370,20 @@ _player addAction [
...
+    -504,
+    true,
+    false,
+    "",
+    "
+        alive _originalTarget
+        && {build_confirmed isEqualTo 0}
+        && {(typeOf ((attachedObjects _target) select 0)) in [KPLIB_b_crateSupply, KPLIB_b_crateAmmo, KPLIB_b_crateFuel]}
+    "
+];
```
Comment (Wyqer) [inline Missionframework/functions/fn_addActionsPlayer.sqf:387]: In general you expect the crate to be always the first attached object. Just thinking about having a IR strobe via ACE attached to you or something similar (backpack on chest, etc.).
Then the crate would be on a different position.
Maybe give it a test with something like
```sqf
!((((attachedObjects _originalTarget) apply {typeOf _x}) arrayIntersect [KPLIB_b_crateSupply, KPLIB_b_crateAmmo, KPLIB_b_crateFuel]) isEqualTo [])
```
Quite messy and may overlook something, but I guess you get what I wanted to point out :)

**Review by Wyqer — DISMISSED** [review]

File: Missionframework/functions/fn_addActionsPlayer.sqf (hunk 4 above)
Comment (Dubjunk) [reply Missionframework/functions/fn_addActionsPlayer.sqf:387]: I'll do some tests on it.

File: Missionframework/functions/fn_addActionsPlayer.sqf (hunk 1 above)
Comment (Dubjunk) [reply Missionframework/functions/fn_addActionsPlayer.sqf:375]: I'm wondering why it doesn't throw any errors while testing 🤔

**Comment by Dubjunk** [comment]
I've made another local test and found some more little bugs, generally it works as intended but there's still a simulation problem when detaching crates. Sometimes they randomly hang up in he air in the position they were dropped, fixes itself when another crate is moved in any way.

**Review by veteran29 — APPROVED** [review]
LGTM

Fixed floating crates and prevented players from putting them inside vehicles or driving while carrying.

**Comment by veteran29** [comment]
Closes #780

**Pull request merged**